lib/hooks/useFacets/utils.js

```diff
@@ -40,6 +40,12 @@ function buildQuery(queryParams, pathComponents, resourceData, logger) {
     queryTemplate = getAdvancedSearchTemplate(queryValue);
   }
 
+  // When switching between Browse and Search some facet data can take a while to update because of async store updates.
+  // In such cases we shouldn't make requests and we can prevent it by checking for undefined query templates.
+  if (!queryTemplate) {
+    return null;
+  }
+
```

Comment:
We should reconsider what the function returns at the end. Let's return just return `cql`, since `undefined` has no meaning anymore.

```
return cql === undefined
    ? CQL_FIND_ALL
    : cql;
    ```